testrunner.py

```diff
@@ -17,12 +18,40 @@ def test_status(self, expected_status, actual_status):
         In a separate function to make debugging easy with py.test
         """
         assert expected_status == actual_status
-        
-    def run_stage(self, stage):
+    
+    def test_log(lines, log_contains):
+        """
+        Checks if a series of log lines contains a regex specified in the
+        output stage. It will flag true on teh first log_contains regex match
+        and then assert on the flag at the end of the function
+        """
+        found = False
+        for line in lines:
+           if log_contains.search(line):
+            found = True
+            break
+        assert(found)
+
+    def run_stage(self, stage, logger_obj):
         """
         Runs a stage in a test by building an httpua object with the stage
         input, waits for output then compares expected vs actual output
         """
         http_ua = http.HttpUA(stage.input)
-        http_ua.send_request()
-        self.test_status(stage.output.status,http_ua.response_object.status)
+        if stage.output.log_contains:
+            start = datetime.datetime.now()
+            http_ua.send_request()
+            end = datetime.datetime.now()
+            logger_obj.set_times(start, end)
+            lines = logger_obj.get_logs()
+            if type(lines) != []:
```

Comment:
Being paranoid here, ill remove this check. The test will fail during the check_log method. Duck typing ftw!